app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryShowActivity.kt

```diff
@@ -80,12 +84,27 @@ class ItineraryShowActivity : BaseMapActivity() {
                 }
             }
         }
+        deleteButton = findViewById(R.id.mission_delete)
+        deleteButton.visibility =
+            if (authService.getCurrentSession()?.user?.uid == ownerUid) View.VISIBLE else View.GONE
 
         goToMissionInProgressButton = findViewById(R.id.buttonToMissionInProgressActivity)
-        goToMissionInProgressButton.isEnabled = droneService.isConnected()
+        canMissionBeLaunched()
+    }
 
-        deleteButton = findViewById(R.id.mission_delete)
-        deleteButton.visibility = if (authService.getCurrentSession()?.user?.uid == ownerUid) View.VISIBLE else View.GONE
+    /**
+     * Check if there is a connected drone, and if the user or the simulation is close enough to launch a mission
+     */
+    private fun canMissionBeLaunched() {
```

Comment:
The name of the function is misleading, maybe have ```goToMissionInProgressButton.isEnabled = canMissionBeLaunched()```

Reply:
Agree, if the method name starts with is/have/can, it usually means that a boolean will be returned